cloudify_cli/commands/install.py

```diff
@@ -0,0 +1,180 @@
+########
+# Copyright (c) 2015 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+"""
+Handles 'cfy install'
+"""
+
+import os
+import urlparse
+
+from cloudify_cli import utils
+from cloudify_cli.commands import blueprints
+from cloudify_cli.commands import deployments
+from cloudify_cli.commands import executions
+from cloudify_cli.constants import DEFAULT_BLUEPRINT_FILE_NAME
+from cloudify_cli.constants import DEFAULT_BLUEPRINT_PATH
+from cloudify_cli.constants import DEFAULT_INPUTS_PATH_FOR_INSTALL_COMMAND
+from cloudify_cli.constants import DEFAULT_INSTALL_WORKFLOW
+from cloudify_cli.exceptions import CloudifyCliError
+
+
+def install(blueprint_path, blueprint_id, validate_blueprint, archive_location,
+            blueprint_filename, deployment_id, inputs, workflow_id, parameters,
+            allow_custom_parameters, timeout, include_logs,
+            auto_generate_ids):
+
+    # First, make sure the `blueprint_path` wasn't supplied with
+    # `archive_location` or with `blueprint_filename`
+    check_for_mutually_exclusive_arguments(blueprint_path,
+                                           archive_location,
+                                           blueprint_filename)
+
+    # The presence of the `archive_location` argument is used to distinguish
+    # between `install` in 'blueprints upload' mode,
+    # and `install` in 'blueprints publish archive' mode.
+    if archive_location:
+
+        blueprints.check_if_archive_type_is_supported(archive_location)
+
+        if not blueprint_filename:
```

Comment:
I changed these parts of the code in accordance with your suggestion. Some changes indeed made the code look better, but some of them made things less readable in my opinion. So, in the sake of consistency, I decided to stick with the old way of things.